### PR TITLE
Support `repo sign` subcommand for manually sign

### PR DESCRIPTION
**Pull request by lucklove**
Signed-off-by: lucklove <gnu.crazier@gmail.com>

### What problem does this PR solve? 

This PR introduce a sub command:
```
tiup repo sign <manifest-file> [key1] [key2]
```

It will append signature to the manifest file

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

```
cat /tmp/repo/timestamp.json
tiup repo init /tmp/repo -i /tmp/keys
tiup repo sign /tmp/repo/timestamp.json /tmp/keys/*-root.json
cat /tmp/repo/timestamp.json
```

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### CODE REVIEW

File: cmd/repo.go [hunk 1]
```diff
@@ -73,6 +75,85 @@ of components or the repository itself.`,
...
+		// Signatures value
+		Signatures []map[string]string `json:"signatures"`
+		// Signed value
+		Signed interface{} `json:"signed"`
+	}
+
+	cmd := &cobra.Command{
+		Use:   "sign <manifest-file> [key-files]",
+		Short: "Add signatures to a manifest file",
+		RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (AstroProfundis) [inline cmd/repo.go:90]: prefer putting the real logic in `manifest/repo.go`

File: cmd/repo.go [hunk 2]
```diff
@@ -73,6 +75,85 @@ of components or the repository itself.`,
...
+				if err != nil {
+					return err
+				}
+
+				sig, err := ki.Signature(payload)
+				if err != nil {
+					return err
+				}
+
+				m.Signatures = append(m.Signatures, map[string]string{
```
Comment (AstroProfundis) [inline cmd/repo.go:136]: I think it's better to check for exist signature fist, if the key already signed this manifest, we can just skip it (or maybe overwrite the signature?)

File: pkg/repository/crypto/keys.go [hunk 3]
```diff
@@ -112,6 +112,7 @@ func (s *keychain) KeyIDs() []string {
 func NewKeyPair(keyType, keyScheme string) (PubKey, PrivKey, error) {
 	// We only support RSA now
 	if keyType != KeyTypeRSA {
+		panic(keyType)
```
Comment (lonng) [inline pkg/repository/crypto/keys.go:115]: ?

File: pkg/repository/v1manifest/repo.go [hunk 4]
```diff
@@ -134,6 +134,81 @@ func GenAndSaveKeys(keys map[string][]*KeyInfo, ty string, num int, dir string)
...
+
+	m := manifestT{}
+	content, err := ioutil.ReadFile(mfile)
+	if err != nil {
+		return err
+	}
+	if err := json.Unmarshal(content, &m); err != nil {
+		return err
+	}
+	payload, err := json.Marshal(m.Signed)
```
Comment (lonng) [inline pkg/repository/v1manifest/repo.go:160]: The `m.Signed` part should be marshaled by `canonicaljson-go`.

**Pull request merged**